addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -15,309 +15,431 @@ odoo.define('website_slides.fullscreen', function (require) {
+        // Public
+        //--------------------------------------------------------------------------
+        updateTitle: function (data) {
+            $('.o_wslides_fs_slide_title').empty().html(QWeb.render('website.course.fullscreen.title', data));
+        },
+        //--------------------------------------------------------------------------
+        // Handler
+        //--------------------------------------------------------------------------
+		_onClickToggleSidebar: function (ev) {
+			this.trigger_up('toggle_sidebar');
+        },
+	});
+
+    var Sidebar = Widget.extend({
+		events: {
+            "click .o_wslides_fs_sidebar_slide_tab": '_onClickTab',
+            "click .o_wslides_slide_quiz": '_onClickMiniQuiz'
+		},
+		init: function (parent, data) {
+			this.slideCompletionMap = {}; // map slide_id -> completion
+			return this._super.apply(this, arguments);
+        },
+        start: function (){
+            this.activeTab = this.$('.o_wslides_fs_sidebar_slide_tab.active');
+            this.activeTabParent = this.$('li.active');
+        },
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+        setCurrentSlide: function (slide_id) {
+            // change the UI
+            this.activeTab.removeClass('active');
+            this.activeTabParent.removeClass('active');
+            this.activeTab = this.$('.o_wslides_fs_sidebar_slide_tab[data-id='+slide_id+']');
+            this.activeTabParent = this.$('li[slide_id='+slide_id+']');
+            this.activeTab.addClass('active');
+            this.activeTabParent.addClass('active');
+        },
+        setSlideCompleted: function (slide_id) {
+            // green up the bullet
+            this.$('#check-'+slide_id).replaceWith($('<i class="check-done o_wslides_slide_completed fa fa-check-circle"></i>'));
+        },
+		toggle: function () {
+            this.$('.o_wslides_fs_sidebar').toggleClass('o_wslides_fs_sidebar_hidden');
+		},
+        updateProgressbar: function (channelCompletion) {
+            var completion = Math.min(100, channelCompletion);
+            $('.o_wslides_fs_sidebar_progress_gauge').css('width', completion + "%" );
+            $('.o_wslides_progress_percentage').text(completion);
+        },
+        //--------------------------------------------------------------------------
+        // Handler
+        //--------------------------------------------------------------------------
+        _onClickMiniQuiz: function (ev){
+            var slideIndex = parseInt($(ev.currentTarget).attr('index'));
+            this.trigger_up('click_miniquiz', {slideIndex: slideIndex});
+        },
+		_onClickTab: function (ev) {
+            // trigger_up the event with slide id
+            var slideIndex = parseInt($(ev.currentTarget).attr('index'));
+			this.trigger_up('change_slide', {slideIndex: slideIndex});
```

Comment:
I think it will be simpler to use slide_id rather than index.

Reply:
Done